cur-lib/cur/stdlib/tactics/standard.rkt

```diff
@@ -18,14 +19,45 @@
  | Tactics should probably not error on failure. Perhaps Maybe monad, or list monad, or return proof
  | state unmodified, or raise exception, or ...
  |#
-(define-tactic (intro name ps)
-  (cur-match (proof-state-current-goal-ref ps)
-    [(forall (x:id : P:expr) body:expr)
-     (let* ([ps (proof-state-extend-env ps name #'P)]
-            [ps (proof-state-current-goal-set ps #'body)]
-            [ps (proof-state-fill-proof-hole ps (lambda (x) #`(λ (#,name : P) #,x)))])
-       ps)]
-    [_ (error 'intro "Can only intro when current goal is of the form (∀ (x : P) body)")]))
+(define-tactic intro
+  (case-lambda
+    [(ps)
```

Comment:
I think it would be nicer to have the proof state as the first argument, use a curried form, or a parameter